File: backend/app/schemas/usage.py
"""Response schemas for the AI usage/cost telemetry endpoint."""

from pydantic import BaseModel, Field


class ModelUsageRow(BaseModel):
    model: str
    provider: str = ""
    calls: int = 0
    input_tokens: int = 0
    output_tokens: int = 0
    cost_usd: float = 0.0


class CallUsageRow(BaseModel):
    call: str
    calls: int = 0
    cost_usd: float = 0.0


class DayUsageRow(BaseModel):
    date: str
    calls: int = 0
    cost_usd: float = 0.0


class UsageSummaryResponse(BaseModel):
    days: int = 30
    total_cost_usd: float = 0.0
    total_input_tokens: int = 0
    total_output_tokens: int = 0
    total_cache_read_tokens: int = 0
    total_calls: int = 0
    by_model: list[ModelUsageRow] = Field(default_factory=list)
    by_call: list[CallUsageRow] = Field(default_factory=list)
    by_day: list[DayUsageRow] = Field(default_factory=list)

File: frontend/lib/api/usage.ts
import { apiRequest } from "./client";
import type { AuthContext } from "./types";

export type ModelUsageRow = {
  model: string;
  provider: string;
  calls: number;
  input_tokens: number;
  output_tokens: number;
  cost_usd: number;
};

export type CallUsageRow = {
  call: string;
  calls: number;
  cost_usd: number;
};

export type DayUsageRow = {
  date: string;
  calls: number;
  cost_usd: number;
};

export type UsageSummary = {
  days: number;
  total_cost_usd: number;
  total_input_tokens: number;
  total_output_tokens: number;
  total_cache_read_tokens: number;
  total_calls: number;
  by_model: ModelUsageRow[];
  by_call: CallUsageRow[];
  by_day: DayUsageRow[];
};

export function getUsageSummary(context: AuthContext, days = 30) {
  return apiRequest<UsageSummary>(`/api/usage/summary?days=${days}`, { context });
}
